source/tests/system/nixnet_utilities.h

```diff
@@ -0,0 +1,202 @@
+#include <gtest/gtest.h>
+#undef interface
+#include <nixnet/nixnet_client.h>
+
+#include <string>
+#include <vector>
+
+#define EXPECT_SUCCESS(response_)    \
+  ([](const auto& response) {              \
+    EXPECT_EQ(0, response.status()); \
+    return response;                 \
+  })(response_)
+
+#define EXPECT_XNET_ERROR(error, response) \
+  if (1) {                                 \
+    EXPECT_EQ(error, (response).status()); \
+  }
+
+namespace nixnet_utilities {
+  
+namespace client = nixnet_grpc::experimental::client;
+namespace pb = google::protobuf;
+using namespace nixnet_grpc;
+
+typedef pb::uint32 u32;
+
+GetPropertyResponse get_property(const client::StubPtr& stub, const nidevice_grpc::Session& session_ref, const client::simple_variant<nixnet_grpc::Property, pb::uint32>& property_id)
+{
+  ::grpc::ClientContext context;
+
+  auto request = GetPropertyRequest{};
+  request.mutable_session()->CopyFrom(session_ref);
+  const auto property_id_ptr = property_id.get_if<nixnet_grpc::Property>();
+  const auto property_id_raw_ptr = property_id.get_if<pb::uint32>();
+  if (property_id_ptr) {
+    request.set_property_id(*property_id_ptr);
+  }
+  else if (property_id_raw_ptr) {
+    request.set_property_id_raw(*property_id_raw_ptr);
+  }
+
+  auto response = GetPropertyResponse{};
+
+  client::raise_if_error(
+      stub->GetProperty(&context, request, &response));
+
+  return response;
+}
+
+inline void _set_property_value(SetPropertyRequest& request, const pb::uint32& value)
+{
+  request.set_u32_scalar(value);
+}
+
+inline void _set_property_value(SetPropertyRequest& request, const bool& value)
+{
+  request.set_bool_scalar(value);
+}
+
+inline void _set_property_value(SetPropertyRequest& request, const std::string& value)
+{
+  request.set_str(value);
+}
+
+inline void _set_property_value(SetPropertyRequest& request, const pb::uint64& value)
+{
+  request.set_u64_scalar(value);
+}
+
+inline void _set_property_value(SetPropertyRequest& request, const pb::int32& value)
+{
+  request.set_i32_scalar(value);
+}
+
+inline void _set_property_value(SetPropertyRequest& request, const double& value)
+{
+  request.set_f64_scalar(value);
+}
+
+inline void _set_property_value(SetPropertyRequest& request, const std::vector<std::string>& value)
+{
+  // TODO: set_string_array expects a single string?!?
```

Comment:
Is this tracked somewhere?

Reply:
yesh, recently we clubbed string_array case with string itself, so not we dont need to deal with it differently. Hence I have removed that part.